clang/lib/CIR/CodeGen/CIRGenExprScalar.cpp

```diff
@@ -1430,6 +1430,28 @@ mlir::Value CIRGenFunction::emitPromotedScalarExpr(const Expr *e,
   return ScalarExprEmitter(*this, builder).Visit(const_cast<Expr *>(e));
 }
 
+mlir::Value CIRGenFunction::emitScalarOrConstFoldImmArg(unsigned ICEArguments,
```

Comment:
This is also being added in https://github.com/llvm/llvm-project/pull/167125

See my comments there about using a range-for at the call sites and passing the argument expression directly to this function.

I'd prefer to have the other PR merged first, then rebase this one to use the definition of this function that will be added there.